firedrake/evaluate.h

```diff
@@ -18,11 +18,11 @@ struct Function {
 	int n_layers;
 
 	/* Coordinate values and node mapping */
-	double *coords;
+	PetscScalar *coords;
```

Comment:
Note that the _implementation_ that uses this stuff will not be complex-safe (since we need to say `PetscRealPart` inside everywhere).

Reply:
I am not that familiar with complex, would replacing `PetscScalar` with `PetscRealPart` just work in both real and complex cases?  I think I will leave this as it is now...

Reply:
One needs to do something:

`PetscScalar *coords` and then replace any use of `coords[...]` with `PetscRealPart(coords[...])`, and possibly replace temporary variables that are declared as either `double` or `PetscScalar` with `PetscReal` I think.